### PR TITLE
Jdk6 jersey2 support

### PR DESCRIPTION
**Pull request by scannerscan**
### PR checklist

- [ ] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [ ] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

 Basically just improving supportJava6 flag on a server side gen. Please review.

supportJava6 flag was added to the server side model classes generators,
licenseInfo ttemplate was added to JavaJaxRS section,
javaJaxRS templates jdk6 suport was added,
shouldOverride method was commented out to allow server classes to be overriden - what if new methods or signatures changes?

### CODE REVIEW

**Comment by wing328** [comment]
@scannerscan thanks for the PR but Shippable CI reports the following issue:
```
[main] INFO io.swagger.codegen.AbstractGenerator - writing file /root/src/github.com/swagger-api/swagger-codegen/samples/server/petstore/jaxrs/jersey1/src/gen/java/io/swagger/model/EnumArrays.java
[main] ERROR io.swagger.codegen.AbstractGenerator - JavaJaxRS/modelEnum.mustache (No such file or directory)
Exception in thread "main" java.lang.RuntimeException: Could not generate model 'EnumClass'
	at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:366)
	at io.swagger.codegen.cmd.Generate.run(Generate.java:223)
	at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:41)
Caused by: java.lang.RuntimeException: can't load template JavaJaxRS/modelEnum.mustache
	at io.swagger.codegen.AbstractGenerator.getTemplateReader(AbstractGenerator.java:63)
	at io.swagger.codegen.DefaultGenerator$5.getTemplate(DefaultGenerator.java:660)
	at com.samskivert.mustache.Mustache$IncludedTemplateSegment.execute(Mustache.java:744)
	at com.samskivert.mustache.Mustache$BlockSegment.executeSegs(Mustache.java:827)
	at com.samskivert.mustache.Mustache$SectionSegment.execute(Mustache.java:852)
	at com.samskivert.mustache.Mustache$BlockSegment.executeSegs(Mustache.java:827)
	at com.samskivert.mustache.Mustache$SectionSegment.execute(Mustache.java:863)
	at com.samskivert.mustache.Mustache$BlockSegment.executeSegs(Mustache.java:827)
	at com.samskivert.mustache.Mustache$SectionSegment.execute(Mustache.java:848)
	at com.samskivert.mustache.Template.executeSegs(Template.java:114)
	at com.samskivert.mustache.Template.execute(Template.java:91)
	at com.samskivert.mustache.Template.execute(Template.java:82)
	at io.swagger.codegen.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:666)
	at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:325)
	... 2 more
```
Would you please take a look?

Let me know if you need help fixing it.

(FYI. Your first commit is not linked to your account)

**Comment by scannerscan** [comment]
sure mate! not sure about your last comment: "Your first commit is not
linked to your account"

**Comment by scannerscan** [comment]
@wing328, please review the fix.

**Comment by wing328** [comment]
@scannerscan if you take a look at https://github.com/swagger-api/swagger-codegen/pull/4064/commits, you will find that the first commit in the list is not linked to a Github account.

Ref: https://github.com/swagger-api/swagger-codegen/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by scannerscan** [comment]
I think it is because on my mac i have a newer git version than on my linux vm. Is it an issue with this? Can we leave as it is?

**Comment by wing328** [comment]
Well that commit won't count towards your contribution in https://github.com/swagger-api/swagger-codegen/graphs/contributors.

I'm ok to leave it as it's.

**Comment by scannerscan** [comment]
Ok, fixing..

**Comment by scannerscan** [comment]
Hmm, seems like the script has done some extras that has not been expected.

**Comment by wing328** [comment]
@scannerscan I would suggest you to submit a new PR instead.
